pkg/operation/botanist/component/kubeapiserver/deployment.go

```diff
@@ -697,6 +703,41 @@ func (k *kubeAPIServer) handleSNISettings(deployment *appsv1.Deployment) {
 	deployment.Spec.Template.Spec.Containers[0].Command = append(deployment.Spec.Template.Spec.Containers[0].Command, fmt.Sprintf("--advertise-address=%s", k.values.SNI.AdvertiseAddress))
 }
 
+func (k *kubeAPIServer) handleTLSSNISettings(deployment *appsv1.Deployment, secretNames []string) error {
+	if len(k.values.SNI.TLS) != len(secretNames) {
+		return fmt.Errorf("number of TLS SNI configs and provided secret names is not equal (%d vs. %d)", len(k.values.SNI.TLS), len(secretNames))
+	}
+
+	for i, sni := range k.values.SNI.TLS {
+		var (
+			volumeName      = fmt.Sprintf("%s%d", volumeNamePathPrefixTLSSNISecret, i)
+			volumeMountPath = fmt.Sprintf("%s%d", volumeMountPathPrefixTLSSNISecret, i)
+			flag            = fmt.Sprintf("--tls-sni-cert-key=%s/tls.crt,%s/tls.key", volumeMountPath, volumeMountPath)
+		)
+
+		if len(sni.DomainPatters) > 0 {
+			flag += ":" + strings.Join(sni.DomainPatters, ",")
+		}
+
+		deployment.Spec.Template.Spec.Containers[0].Command = append(deployment.Spec.Template.Spec.Containers[0].Command, flag)
+		deployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(deployment.Spec.Template.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
+			Name:      volumeName,
+			MountPath: volumeMountPath,
+			ReadOnly:  true,
+		})
+		deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{
+			Name: volumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: secretNames[i],
```

Comment:
Hm, I have a bad feeling to rely on index equality of two slices. Can we use a helper struct (returned by `reconcileTLSSNISecrets`) to pass the required information here?

Reply:
What are your concerns? I think I have added unit tests for all possible scenarios, so I think we should be good. 👀

Reply:
Done with [0d75548](https://github.com/gardener/gardener/pull/7518/commits/0d75548a338c2fb7e31a99c147bbc8777e48b9db)